src/components/pages/userCoursesListPage.tsx

```diff
@@ -1,7 +1,99 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { UserCourse, Course } from 'devu-shared-modules'
+
 
 import PageWrapper from 'components/shared/layouts/pageWrapper'
+import UserCourseListItem from 'components/listItems/userCourseListItem'
+import Dropdown from 'components/shared/inputs/dropdown'
+
+import RequestService from 'services/request.service'
+
+import styles from './userCoursesListPage.scss'
+import LoadingOverlay from 'components/shared/loaders/loadingOverlay'
+
+const filterOptions = [
+  { label: 'All', value: 'all' },
+  { label: 'Active', value: 'active' },
+  { label: 'Inactive', value: 'inactive' },
+  { label: 'Dropped', value: 'dropped' },
+]
+
+const UserCoursesListPage = () => {
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState(null)
+  const [userCourses, setUserCourses] = useState(new Array<UserCourse>())
+  const [courses, setCourses] = useState<Record<string, Course>>({})
+  const currentFilter = JSON.parse(localStorage.getItem('status') || '')
+  const [filter, setFilter] = useState<'all' | 'active' | 'inactive' | 'dropped'>(currentFilter)
+
+  const fetchData = async () => {
+    try {
+      // The filter isn't implemented by the API yet
+      const userCourses: UserCourse[] = await RequestService.get(`/api/user-courses?filterBy=${filter}`)
+      const courseRequests = userCourses.map((u) => RequestService.get(`/api/courses/${u.courseId}`))
+      const courses: Course[] = await Promise.all(courseRequests)
+
+      localStorage.setItem('status', JSON.stringify(filter) || '')
+
+      // Mapify course ids so we can look them up more easilly via their id
+      const courseMap: Record<string, Course> = {}
+      for (const course of courses) courseMap[course.id || ''] = course
+
+      setUserCourses(userCourses)
+      setCourses(courseMap)
+    } catch (error) {
+      setError(error)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [filter])
+
+  if (loading) return <LoadingOverlay />
+
+  type Props = {
+    error: Error | null
```

Comment:
don't allow this to be null. No reason to since we should always be passing it an error object.

Reply:
How should I fix this? "error" has an initial state of null, it seems like TS doesn't know for certain that it won't be null (even though it's filtered out by the conditional it still marks it as being a problem)

Reply:
Hmmmmm so the way to fix it would be when returning the error component check if error is truthy before passing it in like this

```javascript
if (error && error !== null) <ErrorPage />
```

But maybe what you have is better cause I don't want to have to do that everywhere. Here's the only thing though. Make sure you check if error is truthy in the error page component though casue otherwise someone could attempt to do `error.message` and get an object undefined. So just have some default error message like "And unknown error has occured" or something. And use error.message if error is not null

Reply:
I just made a commit to solve this issue (In errorPage.tsx) and tested it in Chromium, was that the implementation you were looking for?